src/hotspot/share/nmt/nmtDCmd.cpp

```diff
@@ -137,8 +137,11 @@ void NMTDCmd::execute(DCmdSource source, TRAPS) {
       output()->print_cr("No detail baseline for comparison");
     }
   } else if (_statistics.value()) {
-    if (check_detail_tracking_level(output())) {
+    NMT_TrackingLevel tracking = MemTracker::tracking_level();
+    if (tracking == NMT_detail || tracking == NMT_summary) {
```

Comment:
I think there is an "is_enabled" method in MemTracker that checks for both states. Please use that.